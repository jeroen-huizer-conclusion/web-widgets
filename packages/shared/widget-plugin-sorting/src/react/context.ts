import { Context, createContext, useContext } from "react";
import { SortStoreHost } from "../stores/SortStoreHost";
import { SortInstruction } from "../types/store";

export interface SortAPI {
    version: 1;
    host: SortStoreHost;
    initSortOrder?: SortInstruction[];
}

const SORT_PATH = "com.mendix.widgets.web.sortable.sortContext";

export function getGlobalSortContext(): Context<SortAPI | null> {
    return ((window as any)[SORT_PATH] ??= createContext<SortAPI | null>(null));
}

export function useSortAPI(): SortAPI {
    const api = useContext(getGlobalSortContext());
    if (api === null) {
        throw new Error("SortAPI is not available. Widget is out of parent context.");
    }
    return api;
}
