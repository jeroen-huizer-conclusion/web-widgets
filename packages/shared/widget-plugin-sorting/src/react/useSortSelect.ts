import { useSetup } from "@mendix/widget-plugin-mobx-kit/react/useSetup";
import { SingleSortController } from "../controllers/SingleSortController";
import { BasicSortStore, SortDirection } from "../types/store";

interface HookProps {
    emptyOptionCaption?: string;
    store: BasicSortStore;
}

interface Option {
    caption: string;
    value: string;
}

interface ControlProps {
    value: string | null;
    options: Option[];
    direction: SortDirection;
    onSelect: (value: string) => void;
    onDirectionClick: () => void;
}

export function useSortSelect(props: HookProps, store: BasicSortStore): ControlProps {
    const ctrl = useSetup(() => new SingleSortController({ store }));

    return {
        value: ctrl.selected,
        options: ctrl.options,
        direction: ctrl.direction,
        onSelect: ctrl.select,
        onDirectionClick: ctrl.toggleDirection
    };
}
