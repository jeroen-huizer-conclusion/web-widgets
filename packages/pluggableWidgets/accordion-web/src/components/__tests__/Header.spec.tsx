import "@testing-library/jest-dom";
import { render, RenderResult } from "@testing-library/react";
import { createElement, PropsWithChildren } from "react";
import { Header, HeaderProps } from "../Header";

describe("Header", () => {
    const defaultHeaderProps: PropsWithChildren<HeaderProps> = {
        heading: "headingThree",
        children: "Text"
    };

    function renderHeader(props: Partial<HeaderProps> = {}): RenderResult {
        return render(<Header {...defaultHeaderProps} {...props} />);
    }

    it("renders h1", () => {
        const header = renderHeader({ heading: "headingOne" });

        expect(header.container).toMatchSnapshot();
    });

    it("renders h2", () => {
        const header = renderHeader({ heading: "headingTwo" });

        expect(header.container).toMatchSnapshot();
    });

    it("renders h3", () => {
        const header = renderHeader();

        expect(header.container).toMatchSnapshot();
    });

    it("renders h4", () => {
        const header = renderHeader({ heading: "headingFour" });

        expect(header.container).toMatchSnapshot();
    });

    it("renders h5", () => {
        const header = renderHeader({ heading: "headingFive" });

        expect(header.container).toMatchSnapshot();
    });

    it("renders h6", () => {
        const header = renderHeader({ heading: "headingSix" });

        expect(header.container).toMatchSnapshot();
    });
});
