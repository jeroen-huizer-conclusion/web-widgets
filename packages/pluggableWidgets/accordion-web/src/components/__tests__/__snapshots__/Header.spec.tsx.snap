// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders h1 1`] = `
<DocumentFragment>
  <h1>
    Text
  </h1>
</DocumentFragment>
`;

exports[`Header renders h2 1`] = `
<DocumentFragment>
  <h2>
    Text
  </h2>
</DocumentFragment>
`;

exports[`Header renders h3 1`] = `
<DocumentFragment>
  <h3>
    Text
  </h3>
</DocumentFragment>
`;

exports[`Header renders h4 1`] = `
<DocumentFragment>
  <h4>
    Text
  </h4>
</DocumentFragment>
`;

exports[`Header renders h5 1`] = `
<DocumentFragment>
  <h5>
    Text
  </h5>
</DocumentFragment>
`;

exports[`Header renders h6 1`] = `
<DocumentFragment>
  <h6>
    Text
  </h6>
</DocumentFragment>
`;
